File: app/src/main/java/com/polok/eubmanagement/model/ModuleData.java
package com.polok.eubmanagement.model;

public class ModuleData {
    String title;
    String link;
    String created_at;
    String key;

    public ModuleData(String title, String link, String created_at, String key) {
        this.title = title;
        this.link = link;
        this.created_at = created_at;
        this.key = key;
    }

    public ModuleData() {}

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getLink() {
        return link;
    }

    public void setLink(String link) {
        this.link = link;
    }

    public String getCreatedAt() {
        return created_at;
    }

    public void setCreatedAt(String date) {
        this.created_at = date;
    }

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public String getNotNullText(String text) {
        if (text != null && !text.isEmpty()) return text;
        else return "N/A";
    }
}

File: app/src/main/java/com/polok/eubmanagement/model/ModuleData.kt
package com.polok.eubmanagement.model

import com.google.gson.annotations.SerializedName

data class ModuleData(
    @SerializedName("title") val title: String? = null,
    @SerializedName("link") var link: String? = null,
    @SerializedName("created_at") val createdAt: String? = null,
    @SerializedName("key") val key: String? = null
)